### PR TITLE
LPS-189331 Remove search.experiences.blueprint.json attribute

### PR DESCRIPTION
**Pull request by gustavolimav**
https://liferay.atlassian.net/browse/LPS-189331

### **Note:**
I needed to refactor one test class to remove totally `search.experiences.blueprint.json` so I added a JSON file based on the JSON that was created in the old test class.

I took the opportunity to change some test class's usages of `search.experiences.blueprint.id` to `search.experiences.blueprint.external.reference.code`.

### CODE REVIEW

**Comment by gustavolimav** [comment]
ci:test:relevant

**Review by peerkar — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/spi/searcher/SXPBlueprintSearchRequestContributor.java [hunk 1]
```diff
@@ -85,107 +103,65 @@ private void _contributeSXPBlueprintId(
...
 				}
-			}
-		}
 
-		if (ArrayUtil.isNotEmpty(runtimeException.getSuppressed())) {
-			if (_log.isWarnEnabled()) {
-				_log.warn(runtimeException);
-			}
-		}
+				SXPBlueprint sxpBlueprint =
```
Comment (peerkar) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/spi/searcher/SXPBlueprintSearchRequestContributor.java:130]: inline:

```
_enhance(searchRequestBuilder, _sxpBlueprintLocalService.fetchSXPBlueprint(sxpBlueprintId));

```

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/spi/searcher/SXPBlueprintSearchRequestContributor.java [hunk 2]
```diff
@@ -63,8 +63,26 @@ private void _contributeSXPBlueprintExternalReferenceCode(
...
+					string);
+
+				for (String sxpBlueprintExternalReferenceCode :
+						sxpBlueprintExternalReferenceCodes) {
+
+					if (Validator.isBlank(sxpBlueprintExternalReferenceCode)) {
+						continue;
+					}
+
+					SXPBlueprint sxpBlueprint =
```
Comment (peerkar) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/spi/searcher/SXPBlueprintSearchRequestContributor.java:76]: please inline:

```
				_enhance(searchRequestBuilder, _sxpBlueprintLocalService.
						fetchSXPBlueprintByExternalReferenceCode(
								sxpBlueprintExternalReferenceCode,
								GetterUtil.getLong(
									searchRequestBuilder.withSearchContextGet(
										SearchContext::getCompanyId))));
```

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/spi/searcher/SXPBlueprintSearchRequestContributor.java [hunk 3]
```diff
@@ -85,107 +103,65 @@ private void _contributeSXPBlueprintId(
...
-			SXPBlueprint sxpBlueprint =
-				_sxpBlueprintLocalService.fetchSXPBlueprint(sxpBlueprintId);
+		if (_log.isDebugEnabled()) {
+			_log.debug("Search experiences blueprint " + sxpBlueprint);
+		}
 
-			if (_log.isDebugEnabled()) {
-				_log.debug("Search experiences blueprint " + sxpBlueprint);
-			}
+		if (sxpBlueprint != null) {
```
Comment (peerkar) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/spi/searcher/SXPBlueprintSearchRequestContributor.java:145]: I think Brian prefers:

```
		if (sxpBlueprint == null) {
			return;
		}
```

**Comment by peerkar** [comment]
@gustavolimav I think the contributor code is now leaner in my opinion 👍  Please take a look at the couple formatting suggestions.

The test class: misunderstanding there, sorry. I was merely meaning preserving the ID tests with the new ERC tests. I didn't mean to remove the ERC ones. Can we have the both ones?

**Comment by gustavolimav** [comment]
> @gustavolimav I think the contributor code is now leaner in my opinion +1 Please take a look at the couple formatting suggestions.
> 
> The test class: misunderstanding there, sorry. I was merely meaning preserving the ID tests with the new ERC tests. I didn't mean to remove the ERC ones. Can we have the both ones?

Those [tests](https://github.com/liferay-search/liferay-portal/pull/1015/commits/1cace3787335d6808445fab38e40899ca88092b8) are not covering it (ERC)?

And those [here](https://github.com/liferay-search/liferay-portal/pull/1015/commits/692a2a1a9b4f6084d36ac98be89e7dae3d268dfd) covering ID?

**Comment by peerkar** [comment]
> Just want to keep the majority using ERC and 1 with ID.

Oh, sorry my bad eyes... Sure, ERC is what will be used. For readability, would it even be possible to isolate the ID use case into a single separate test?

**Comment by gustavolimav** [comment]
@peerkar, I isolated it in a new test

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
Hey @peerkar, can you review it just one more time so I can forward it?

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/spi/searcher/SXPBlueprintSearchRequestContributor.java [hunk 4]
```diff
@@ -64,8 +63,25 @@ private void _contributeSXPBlueprintExternalReferenceCode(
 			String string = (String)object;
 
 			if (!Validator.isBlank(string)) {
```
Comment (peerkar) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/spi/searcher/SXPBlueprintSearchRequestContributor.java:65]: I'm pretty sure Brian will want this and every similar instance like:

```
if (Validator.isBlank(string)) {
   return;
}

...


```

File: modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/spi/searcher/SXPBlueprintSearchRequestContributor.java [hunk 5]
```diff
@@ -86,132 +102,57 @@ private void _contributeSXPBlueprintId(
...
-		try {
-			if (Validator.isNotNull(sxpBlueprintJSON)) {
-				_sxpBlueprintSearchRequestEnhancer.enhance(
-					searchRequestBuilder, sxpBlueprintJSON);
-			}
-		}
-		catch (Exception exception) {
-			runtimeException.addSuppressed(exception);
-		}
+		if (sxpBlueprintIds != null) {
```
Comment (peerkar) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/spi/searcher/SXPBlueprintSearchRequestContributor.java:123]: The same here:

```

if (sxpBlueprintIds == null) {
   return;

```

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by gustavolimav** [comment]
ci:test:relevant

**Comment by gustavolimav** [comment]
ci:test:search

**Comment by gustavolimav** [comment]
Pull Request forwarded: https://github.com/brianchandotcom/liferay-portal/pull/138912

**Pull request closed without merging**